### PR TITLE
Update blazegraph documentation

### PR DESCRIPTION
**Pull request by imsdu**
Screenshot_2021-05-17 Views · Blue Brain Nexus

### CODE REVIEW

**Comment by imsdu** [comment]
Screenshot_2021-05-12 SparqlView · Blue Brain Nexus

**Comment by imsdu** [comment]
Screenshot_2021-05-12 AggregateSparqlView · Blue Brain Nexus

**Comment by umbreak** [comment]
On the views page, `SparqlView`, `ElasticSearchView`, (etc) sections don't seem to be `###` (because. they don't show up on the table of contents. They should. Also the order of their appearance should be the same as the order on the left side panel, whatever order we choose. I would recommend:

- Elasticsearch view
- Aggregate elasticsearch view
- Sparql view
- Aggregate sparql view
- Composite view

File: docs/src/main/paradox/docs/delta/api/current/views/aggregated-sparql-view-api.md [hunk 1]
```diff
@@ -45,13 +45,167 @@ In order to have a more robust SPARQL aggregation support, please make us of Com
...
+
+## Fetch
+
+```
+GET /v1/views/{org_label}/{project_label}/{view_id}?rev={rev}&tag={tag}
+```
 
-### SPARQL query
+where ...
+- `{rev}`: Number - the targeted revision to be fetched. This field is optional and defaults to the latest revision.
```
Comment (umbreak) [inline docs/src/main/paradox/docs/delta/api/current/views/aggregated-sparql-view-api.md:177]: you need an empty line here, in order to have a proper formatting

File: docs/src/main/paradox/docs/delta/api/current/views/aggregated-sparql-view-api.md [hunk 2]
```diff
@@ -45,13 +45,167 @@ In order to have a more robust SPARQL aggregation support, please make us of Com
...
+Response
+:   @@snip [fetched.json](../assets/views/blazegraph/aggregate/fetched.json)
+
+## Fetch original payload
+
+```
+GET /v1/views/{org_label}/{project_label}/{view_id}/source?rev={rev}&tag={tag}
+```
+where ...
+- `{rev}`: Number - the targeted revision to be fetched. This field is optional and defaults to the latest revision.
```
Comment (umbreak) [inline docs/src/main/paradox/docs/delta/api/current/views/aggregated-sparql-view-api.md:195]: you need an empty line here, in order to have a proper formatting

File: docs/src/main/paradox/docs/delta/api/current/views/aggregated-sparql-view-api.md [hunk 3]
```diff
@@ -72,8 +226,19 @@ The `Content-Type` HTTP header for POST request is `application/sparql-query`.
...
-:   @@snip [sparql-view-search.sh](../assets/views/sparql-view-search.sh)
+:   @@snip [search.sh](../assets/views/blazegraph/search.sh)
 
 Response
-:   @@snip [sparql-view-search.json](../assets/views/sparql-view-search.json)
+:   @@snip [search.json](../assets/views/blazegraph/search.json)
+
+@@@ note { .tip title="Response formats" }
+
+From Delta 1.5, we have added support for multiple Content Negotiation types when querying the SPARQL view, allowing clients to request different response formats. The Content Negotiation is controlled by the HTTP `Accept` header. The following values are supported:
```
Comment (umbreak) [inline docs/src/main/paradox/docs/delta/api/current/views/aggregated-sparql-view-api.md:236]: I don't think we need this in a `note` block but just in the previous paragraph where says `The Content-Type HTTP Header`.

Just adding block notes on every release is not going to make the documentation more understandable.

**Review by umbreak — CHANGES_REQUESTED** [review]

**Review by umbreak — CHANGES_REQUESTED** [review]
On the sparqlview page, the `example` payload contains the `"@type"` as an array. Please simplify this as `"@type": "SparqlView"`

File: docs/src/main/paradox/docs/delta/api/current/views/sparql-view-api.md [hunk 4]
```diff
@@ -59,13 +75,174 @@ The resulting RDF triples will contain the resources metadata.
...
 
-### SPARQL query
+## Fetch
+
+```
+GET /v1/views/{org_label}/{project_label}/{view_id}?rev={rev}&tag={tag}
+```
+
+where ...
+- `{rev}`: Number - the targeted revision to be fetched. This field is optional and defaults to the latest revision.
```
Comment (umbreak) [inline docs/src/main/paradox/docs/delta/api/current/views/sparql-view-api.md:215]: empty line required

File: docs/src/main/paradox/docs/delta/api/current/views/sparql-view-api.md [hunk 5]
```diff
@@ -59,13 +75,174 @@ The resulting RDF triples will contain the resources metadata.
...
+Response
+:   @@snip [fetched.json](../assets/views/blazegraph/sparql/fetched.json)
+
+## Fetch original payload
+
+```
+GET /v1/views/{org_label}/{project_label}/{view_id}/source?rev={rev}&tag={tag}
+```
+where ...
+- `{rev}`: Number - the targeted revision to be fetched. This field is optional and defaults to the latest revision.
```
Comment (umbreak) [inline docs/src/main/paradox/docs/delta/api/current/views/sparql-view-api.md:233]: empty line required

File: docs/src/main/paradox/docs/delta/api/current/views/sparql-view-api.md [hunk 6]
```diff
@@ -85,13 +262,24 @@ The `Content-Type` HTTP header for POST request is `application/sparql-query`.
...
 
 Request
-:   @@snip [sparql-view-search.sh](../assets/views/sparql-view-search.sh)
+:   @@snip [search.sh](../assets/views/blazegraph/search.sh)
 
 Response
-:   @@snip [sparql-view-search.json](../assets/views/sparql-view-search.json)
+:   @@snip [sparql-view-search.json](../assets/views/blazegraph/search.json)
+
+@@@ note { .tip title="Response formats" }
```
Comment (umbreak) [inline docs/src/main/paradox/docs/delta/api/current/views/sparql-view-api.md:270]: Again, I don't think that needs to be inside a note block

**Review by umbreak — CHANGES_REQUESTED** [review]

File: docs/src/main/paradox/docs/delta/api/current/views/sparql-view-api.md [hunk 7]
```diff
@@ -116,7 +304,7 @@ where...
 - `lastProcessedEventDateTime` - timestamp of the last event processed by the view
 - `delayInSeconds` - number of seconds between the last processed event timestamp and the last known event timestamp
  
-### Restart view
+## Restart indexing
```
Comment (umbreak) [inline docs/src/main/paradox/docs/delta/api/current/views/sparql-view-api.md:307]: If you want to call it `Restart indexing` here (which I agree), then let's also call it `Restart indexing` in elasticsearch (currently is called `restart view`)

**Review by umbreak — CHANGES_REQUESTED** [review]
It is also missing a `Fetch offset` section, as in elasticsearch. BTW, I would then rename it to `Fetch indexing` to align it with `Restart indexing`

File: docs/src/main/paradox/docs/delta/api/current/assets/views/deprecate.sh [hunk 8]
```diff
@@ -0,0 +1,2 @@
+curl -X DELETE \
+     "https://nexus.example.com/v1/views/myorg/myproj/nxv:myview?rev=5"
```
Comment (umbreak) [inline docs/src/main/paradox/docs/delta/api/current/assets/views/deprecate.sh:2]: I think it should be `?rev=2` if you follow the examples sequentially.
Same for the elasticsearch deprecate example

**Review by umbreak — CHANGES_REQUESTED** [review]

File: docs/src/main/paradox/docs/delta/api/current/assets/views/deprecate.sh (hunk 8 above)
Comment (imsdu) [reply docs/src/main/paradox/docs/delta/api/current/assets/views/deprecate.sh:2]: It is `?rev=3` as there is an update and tagging operations before

File: docs/src/main/paradox/docs/delta/api/current/assets/views/blazegraph/sparql/payload.json [hunk 9]
```diff
@@ -0,0 +1,13 @@
+{
+  "@id": "https://bluebrain.github.io/nexus/vocabulary/indexing-view"
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/assets/views/blazegraph/sparql/payload.json:2]: missing a comma

File: docs/src/main/paradox/docs/delta/api/current/assets/views/blazegraph/sparql/payload.json [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+  "@type": "SparqlView",
+  "resourceSchemas": [
+    "https://bluebrain.github.io/nexus/vocabulary/some-schema"
+  ],
+  "resourceTypes": [
+    "https://bluebrain.github.io/nexus/vocabulary/SomeType"
+  ],
+  "includeMetadata": false,
+  "includeDeprecated": false,
+  "permission": "my/permission",
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/assets/views/blazegraph/sparql/payload.json:12]: extra comma

File: docs/src/main/paradox/docs/delta/api/current/assets/views/blazegraph/sparql/fetched.json [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+  "@id": "https://bluebrain.github.io/nexus/vocabulary/myview",
+  "@type" : [
+    "SparqlView",
+    "View"
+  ],
+  "_constrainedBy": "https://bluebrain.github.io/nexus/schemas/views.json",
+  "_createdAt": "2021-04-18T17:10:22.748Z",
+  "_createdBy": "http://localhost/v1/realms/myrealm/users/Bob",
+  "_deprecated": false,
+  "_self": "http://localhost/v1/views/org/proj/nxv:myview",
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/assets/views/blazegraph/sparql/fetched.json:15]: the links do not seem to match the environment, I was expecting `http://localhost:8080/v1`

File: docs/src/main/paradox/docs/delta/api/current/assets/views/blazegraph/list.json [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+      "_createdBy": "http://localhost:8080/v1/realms/myrealm/users/john",
+      "_updatedAt": "2021-04-18T17:10:22.748Z",
+      "_updatedBy": "http://localhost:8080/v1/realms/myrealm/users/john"
+    },
+    {
+      "@id": "https://bluebrain.github.io/nexus/vocabulary/myview2",
+      "@type": [
+        "https://bluebrain.github.io/nexus/vocabulary/ElasticSearchView",
+        "https://bluebrain.github.io/nexus/vocabulary/View",
+        "https://bluebrain.github.io/nexus/vocabulary/Alpha"
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/assets/views/blazegraph/list.json:50]: this looks odd

File: docs/src/main/paradox/docs/delta/api/current/assets/views/blazegraph/sparql/fetched.json (hunk 11 above)
Comment (bogdanromanx) [reply docs/src/main/paradox/docs/delta/api/current/assets/views/blazegraph/sparql/fetched.json:15]: this seems to be the case in a lot of examples...

File: docs/src/main/paradox/docs/delta/api/current/views/aggregated-sparql-view-api.md [hunk 13]
```diff
@@ -45,13 +45,188 @@ In order to have a more robust SPARQL aggregation support, please make us of Com
...
+## Create using POST
+
+```
+POST /v1/view/{org_label}/{project_label}
+  {...}
+```
+
+The json payload:
+
+- If the `@id` value is found on the payload, this @id will be used.
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/views/aggregated-sparql-view-api.md:57]: at back-ticks to `@id`

File: docs/src/main/paradox/docs/delta/api/current/views/aggregated-sparql-view-api.md [hunk 14]
```diff
@@ -45,13 +45,188 @@ In order to have a more robust SPARQL aggregation support, please make us of Com
...
+Payload
+:   @@snip [payload.json](../assets/views/blazegraph/aggregate/payload.json)
+
+Response
+:   @@snip [created.json](../assets/views/blazegraph/aggregate/created.json)
+
+## Create using PUT
+
+This alternative endpoint to create a view is useful in case the json payload does not contain an `@id` but you want
+to specify one. The @id will be specified in the last segment of the endpoint URI.
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/views/aggregated-sparql-view-api.md:75]: back-ticks

File: docs/src/main/paradox/docs/delta/api/current/views/sparql-view-api.md [hunk 15]
```diff
@@ -49,23 +65,204 @@ The resulting RDF triples will contain the resources metadata.
...
+## Create using POST
+
+```
+POST /v1/view/{org_label}/{project_label}
+  {...}
+```
+
+The json payload:
+
+- If the `@id` value is found on the payload, this @id will be used.
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/views/sparql-view-api.md:85]: back-ticks

File: docs/src/main/paradox/docs/delta/api/current/views/sparql-view-api.md [hunk 16]
```diff
@@ -49,23 +65,204 @@ The resulting RDF triples will contain the resources metadata.
...
+Payload
+:   @@snip [payload.json](../assets/views/blazegraph/sparql/payload.json)
+
+Response
+:   @@snip [created.json](../assets/views/blazegraph/sparql/created.json)
+
+## Create using PUT
+
+This alternative endpoint to create a view is useful in case the json payload does not contain an `@id` but you want
+to specify one. The @id will be specified in the last segment of the endpoint URI.
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/views/sparql-view-api.md:103]: back-ticks

**Review by bogdanromanx — CHANGES_REQUESTED** [review]

**Review by umbreak — APPROVED** [review]

**Review by bogdanromanx — APPROVED** [review]

**Pull request merged**